packages/persist/lib/server.integration.test.ts

```diff
@@ -0,0 +1,207 @@
+                        records: [],
+                        providerConfigKey: seed.connection.provider_config_key,
+                        connectionId: seed.connection.connection_id,
+                        lastSyncDate: new Date(),
+                        trackDeletes: false,
+                        softDelete: true
+                    }),
+                    headers: {
+                        'Content-Type': 'application/json'
+                    }
+                }
+            );
+            expect(response.status).toEqual(400);
+            const respBody = (await response.json()) as any[];
+            expect(respBody[0]['errors']['issues'][0]['path'][0]).toEqual('records');
+            expect(respBody[0]['errors']['issues'][0]['message']).toEqual('Array must contain at least 1 element(s)');
+        });
+
+        it('should save records', async () => {
+            const model = 'MyModel';
+            const records = [
+                { id: 1, name: 'r1' },
+                { id: 2, name: 'r2' }
+            ];
+            const response = await fetch(
+                `${serverUrl}/environment/${seed.env.id}/connection/${seed.connection.id}/sync/${seed.sync.id}/job/${seed.syncJob.id}/records`,
+                {
+                    method: 'POST',
+                    body: JSON.stringify({
+                        model,
+                        records: records,
+                        providerConfigKey: seed.connection.provider_config_key,
+                        connectionId: seed.connection.connection_id,
+                        activityLogId: seed.activityLogId,
+                        lastSyncDate: new Date(),
+                        trackDeletes: false
+                    }),
+                    headers: {
+                        'Content-Type': 'application/json'
+                    }
+                }
+            );
+            expect(response.status).toEqual(201);
+        });
+    });
+
+    it('should delete records ', async () => {
+        const model = 'MyModel';
+        const records = [
+            { id: 1, name: 'r1' },
+            { id: 2, name: 'r2' }
+        ];
+        const response = await fetch(
+            `${serverUrl}/environment/${seed.env.id}/connection/${seed.connection.id}/sync/${seed.sync.id}/job/${seed.syncJob.id}/records`,
+            {
+                method: 'DELETE',
+                body: JSON.stringify({
+                    model,
+                    records: records,
+                    providerConfigKey: seed.connection.provider_config_key,
+                    connectionId: seed.connection.connection_id,
+                    activityLogId: seed.activityLogId,
+                    lastSyncDate: new Date(),
+                    trackDeletes: false
+                }),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            }
+        );
+        expect(response.status).toEqual(201);
```

Comment:
not related to this PR: this test do not actually check if the records were deleted